soc/arm/nordic_nrf/nrf52/soc.c

```diff
@@ -47,6 +47,22 @@ extern void z_arm_nmi_init(void);
 #define LOG_LEVEL CONFIG_SOC_LOG_LEVEL
 LOG_MODULE_REGISTER(soc);
 
+int soc_mcuboot_mode_set(uint8_t mode)
+{
+	nrf_power_gpregret2_set(NRF_POWER, mode);
```

Comment:
From spot-checking a product specification, the GPREGRET and GPREGRET2 registers seem to be available for application developer use too. Is there any risk in using them for this purpose?

Reply:
GPREGRET is used by NCS internally for instance, it is also place where nRF SoC stores reboot-type in zephyr.

Reply:
I can use GPREGRET for transfer both the reboot-type and the recovery-dfu request. This implies that I need to introduce an API for retrieve reboot-type as well.

Reply:
If you see usage of GPREGRET2 as the waste of resources I can make a better solution:

The idea would be to start using GPREGRET for request of recovery-dfu as well (along with reboot-type to keep of which it is used currently).
As intermediate steep I would implement API for retrieve  reboot-type as there is `sys_arch_reboot(int type)` call but there is no a counterpart to it which would allows to check what a reboot-type was used.
Of course this mean that zephyr common API for get reboot type will be needed:
  there is `void sys_reboot(int type)` but zephyr doesn't have API for retrieve the `type`

@mbolivar-nordic What is your opinion on that ^^ idea?

Reply:
alternative: https://github.com/zephyrproject-rtos/zephyr/pull/46084